nbviewer/providers/base.py

```diff
@@ -442,27 +464,6 @@ def write_error(self, status_code, **kwargs):
     # response caching
     #---------------------------------------------------------------
 
-    @property
```

Comment:
I think it's better to leave the cache-related things here in the cache-related section, so that related items are together. Can we leave most, if not all of these alone, or at least the ones that have logical groupings?

Reply:
I agree with your reasoning and will make those changes soon, when the time comes up.

I was originally waffling about whether or not to organize them that way, since I couldn't decide for myself what seemed most "consistent".

Reply:
It's hard to choose! I think logical groupings are useful, but they can make it hard to intuitively scroll to your destination since you have to know something's in a group, as I'm sure you've found. It does make it easier to find nearby relevant things, though, so it's a trade off.

I don't think the bulk explicit property->settings key is super sustainable. Maybe overriding `__getattr__` to look in settings for most things by default is better?

Reply:
I wouldn't have been able to express it in those terms/so eloquently myself, but yes that was exactly the choice I was confused about. With the logical groupings there also seemed to be the issue that some of the ones that are currently in the giant list of properties probably should belong to their own groups, or some of the existing groups, and then which ones belong to existing groups? What would the new groups be? Should there be a formal layer of abstraction encapsulating those groups, and if so what would be the best choice?

Anyway I am looking into putting back the ones which were originally in logical groups and leaving it at that for now.

I also 100% agree that the bulk explicit property -> settings key isn't very sustainable. My only concern with overriding `__getattr__` is that might lead to an ignorant/naive person like me in the future looking at the code and seeing the `__getattr__` calls and assuming that it's the built-in version, and then writing code based on that assumption, and then the code doesn't work and then they're confused and upset and don't understand why it isn't working the way it's supposed to, yadda, yadda.

Although I suppose maybe something which does a `try`/`except` block, where it first tries the Python built-in version, and then if it doesn't work then it looks in settings, could work and avoid the above issue? I agree that having to fish everything out of the Tornado handler's settings attribute isn't actually efficient.